support-metrics-common/src/main/java/io/confluent/support/metrics/submitters/KafkaSubmitter.java

```diff
@@ -110,11 +110,11 @@ protected void submit(byte[] bytes, Producer<byte[], byte[]> producer) {
 
   private Producer<byte[], byte[]> createProducer() {
     Properties props = new Properties();
-    if (zkUtils == null) {
-      zkUtils = zkUtilsProvider.zkUtils();
+    if (zkClient == null) {
+      zkClient = zkClientProvider.zkClient();
```

Comment:
Since we're using `KafkaZkClient` now, don't we need some close logic?

Reply:
Like in the other comment, we don't need to close it as it will be closed when `KafkaServer.shutdown` is called.

Reply:
To clarify, does that mean that the client is a cached instance and not the caller's responsibility to close? Maybe we should document that in `ZkClientProvider`?

Reply:
Yes, it is the same instance used by the broker, so the lifecycle is managed by the broker. This PR doesn't change that. `KafkaZkClient` is used in the exact same way as `ZkUtils` before it. I can add a note to `ZkClientProvider`. It would be better if the interface was more general and did not assume so much, but I don't want to tackle that refactoring in this PR (which is just about fixing the build).